pkg/porter/lifecycle.go

```diff
@@ -102,3 +104,19 @@ func (p *Porter) prepullBundleByReference(opts *BundleActionOptions) error {
 
 	return nil
 }
+
+func slugify(value string) string {
+	var slug string
+	for _, v := range value {
+		if unicode.IsLetter(v) {
+			slug += string(unicode.ToLower(v))
+		}
+		if unicode.IsDigit(v) {
+			slug += string(v)
+		}
+		if unicode.IsSpace(v) {
+			slug += "-"
+		}
```

Comment:
Perhaps we should add a case to allow `-` chars in as well?

Reply:
I think there are even more characters allowed in the name. The biggest limiting factor in what's okay in the installation name being able to reference and store it properly. I recommend that we use the allowed characters that we are adopting for the installation state spec.

> The name segment is required and MUST be 63 characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.

This helps us avoid having names that cannot be stored in certain formats, like kubernetes or as an index value in a database.